### PR TITLE
P2P:Must support bidirectional channel for peer

### PR DESCRIPTION
**Pull request by lochjin**
To connect private qitmeer nodes in the LAN, you can use the configuration parameters: `--lanpeers=[Peer ID]`

### CODE REVIEW

File: p2p/synch/sync.go [hunk 1]
```diff
@@ -7,6 +7,7 @@ package synch
 import (
 	"context"
 	"fmt"
+	"github.com/Qitmeer/qitmeer/core/blockdag"
```
Comment (dindinw) [inline p2p/synch/sync.go:10]: I am not quite understand why we need to add the `blockdag` package dependency if it's just an empty struct marker.
```
// This struct is empty
type Empty struct{}
```

File: config/config.go [hunk 2]
```diff
@@ -90,6 +90,7 @@ type Config struct {
 	RelayNode      string   `long:"relaynode" description:"The address of relay node that routes traffic between two peers over a qitmeer “relay” peer."`
 	Whitelist      string   `long:"whitelist" description:"Add an IP network or IP that will not be banned. (eg. 192.168.1.0/24 or ::1)"`
```
Comment (dindinw) [inline config/config.go:91]: - Can the `LANPeers` and `WhiteList` to be merged into single name `WhiteList` for a better naming and simplify the usage. let the `WhiteList` to support IP address and `PeerID` both. 
- BTW, I don't see the `Whitelist` is used throughout the PR, What happens if `Whitelist` and `LANPeer` both configured.
- I think its good idea to keep using single input `WhiteList` at the first place to minimize the future inconsistence.

File: p2p/synch/sync.go (hunk 1 above)
Comment (lochjin) [reply p2p/synch/sync.go]: > I am not quite understand why we need to add the `blockdag` package dependency if it's just an empty struct marker.
> 
> ```
> // This struct is empty
> type Empty struct{}
> ```

https://github.com/Qitmeer/qitmeer/pull/559/commits/ccc17135aeb81f56305e4cf8ac4b993eb4c65b9f

File: config/config.go (hunk 2 above)
Comment (lochjin) [reply config/config.go]: https://github.com/Qitmeer/qitmeer/pull/559/commits/ffabda6b9d0ec67de105b1fd9d87a71a0fdb5393

**Pull request merged**